src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/DSL.kt

```diff
@@ -68,7 +75,7 @@ class TestData(private val repository: Repository, private val courseAllocationB
     birthDate: LocalDate,
     genderCode: String,
     dsl: OffenderDsl.() -> Unit,
-  ): Offender = repository.save(
+  ): Offender = repository!!.save(
     OffenderBuilder(nomsId, lastName, firstName, birthDate, genderCode, repository = repository, courseAllocationBuilderFactory = courseAllocationBuilderFactory).apply(dsl),
   )
```

Comment:
As `Offender` is the root entity in its entity graph we can't use the new DSL methodolgy with `CourseAllocation` or its children - we'll have to do a full entity graph at the same time. (which will be painful 😢 )